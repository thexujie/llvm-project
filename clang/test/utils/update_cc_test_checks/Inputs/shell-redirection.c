/// Check that shell redirections in the RUN line are handled
// RUN: %clang_cc1 -triple=x86_64-unknown-linux-gnu -emit-llvm < %s 2>/dev/null | FileCheck %s
// RUN: %clang_cc1 -triple=x86_64-unknown-linux-gnu -emit-llvm \
// RUN:   -disable-O0-optnone -o - %s 2>&1 | opt -S -passes=mem2reg \
// RUN:   | FileCheck %s --check-prefix=MEM2REG

int test(int a, int b) {
  return a + b;
}
