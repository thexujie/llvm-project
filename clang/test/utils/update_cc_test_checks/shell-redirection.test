## Check that update_cc_test_checks can parse RUN lines with shell redirection
## This currently fails with errors such as:
## error: error reading '<': No such file or directory
## error: error reading '2>/dev/null': No such file or directory
# RUN: cp -f %S/Inputs/shell-redirection.c %t.c && %update_cc_test_checks %t.c
# RUN: diff -u %t.c %S/Inputs/shell-redirection.c.expected

