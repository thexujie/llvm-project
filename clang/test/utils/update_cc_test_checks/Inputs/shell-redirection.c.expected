// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py
/// Check that shell redirections in the RUN line are handled
// RUN: %clang_cc1 -triple=x86_64-unknown-linux-gnu -emit-llvm < %s 2>/dev/null | FileCheck %s
// RUN: %clang_cc1 -triple=x86_64-unknown-linux-gnu -emit-llvm \
// RUN:   -disable-O0-optnone -o - %s 2>&1 | opt -S -passes=mem2reg \
// RUN:   | FileCheck %s --check-prefix=MEM2REG

// CHECK-LABEL: @test(
// CHECK-NEXT:  entry:
// CHECK-NEXT:    [[A_ADDR:%.*]] = alloca i32, align 4
// CHECK-NEXT:    [[B_ADDR:%.*]] = alloca i32, align 4
// CHECK-NEXT:    store i32 [[A:%.*]], ptr [[A_ADDR]], align 4
// CHECK-NEXT:    store i32 [[B:%.*]], ptr [[B_ADDR]], align 4
// CHECK-NEXT:    [[TMP0:%.*]] = load i32, ptr [[A_ADDR]], align 4
// CHECK-NEXT:    [[TMP1:%.*]] = load i32, ptr [[B_ADDR]], align 4
// CHECK-NEXT:    [[ADD:%.*]] = add nsw i32 [[TMP0]], [[TMP1]]
// CHECK-NEXT:    ret i32 [[ADD]]
//
// MEM2REG-LABEL: @test(
// MEM2REG-NEXT:  entry:
// MEM2REG-NEXT:    [[ADD:%.*]] = add nsw i32 [[A:%.*]], [[B:%.*]]
// MEM2REG-NEXT:    ret i32 [[ADD]]
//
int test(int a, int b) {
  return a + b;
}
